src/test/scala/mesosphere/marathon/core/launcher/impl/OfferProcessorImplTest.scala

```diff
@@ -0,0 +1,117 @@
+package mesosphere.marathon.core.launcher.impl
+
+import com.codahale.metrics.MetricRegistry
+import mesosphere.marathon.core.base.{ Clock, ConstantClock }
+import mesosphere.marathon.core.launcher.{ OfferProcessor, OfferProcessorConfig, TaskLauncher }
+import mesosphere.marathon.core.matcher.base.OfferMatcher
+import OfferMatcher.{ MatchedTasks, TaskLaunchSource, TaskWithSource }
+import mesosphere.marathon.core.matcher.base.OfferMatcher
+import mesosphere.marathon.metrics.Metrics
+import mesosphere.marathon.state.Timestamp
+import mesosphere.marathon.{ MarathonSpec, MarathonTestHelper }
+import org.apache.mesos.Protos.TaskInfo
+import org.mockito.Mockito.{ verify, when }
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future }
+
+class OfferProcessorImplTest extends MarathonSpec {
+  private[this] val offer = MarathonTestHelper.makeBasicOffer().build()
+  private[this] val offerId = offer.getId
+  private[this] val taskInfo1 = MarathonTestHelper.makeOneCPUTask("taskid1").build()
+  private[this] val taskInfo2 = MarathonTestHelper.makeOneCPUTask("taskid2").build()
+  private[this] val tasks = Seq(taskInfo1, taskInfo2)
+
+  test("match successful, launch tasks successful") {
+    val dummySource = new DummySource
+    val tasksWithSource = tasks.map(TaskWithSource(dummySource, _))
+
+    val offerProcessor = createProcessor()
+
+    val deadline: Timestamp = clock.now() + 1.second
+    when(offerMatcher.matchOffer(deadline, offer)).thenReturn(
+      Future.successful(MatchedTasks(offerId, tasksWithSource))
+    )
+    when(taskLauncher.launchTasks(offerId, tasks)).thenReturn(true)
+
+    Await.result(offerProcessor.processOffer(offer), 1.second)
```

Comment:
Yes, but I do like Await.result better. Why use something different in tests than you use in production code?